### DISCUSSION TITLE
Reccurring event series listing

### DISCUSSION THREAD

**Discussion by boundingstar** [koalagator/koalagator#23] [Ideas / Feature Requests]
Roughly half of the events I have found to list are recurring events with a mostly consistent repetition schedule. It would save a lot of effort to be able to list the event once as a series and update individual occurrences as needed to reflect once-off changes to the event, the example I have in mind is event series in Outlook. Perhaps this would require each occurrence to be approved when it is automatically created, to ensure it matches the event source, and avoid recurring events filling the 'future events' list

**Comment by evolve2k** [comment]
I agree there's some unresolved issues around adding repeating events. I'm also cautious as the quantity of a specific regular event can sort of swamp out a once-off event that all things being equal is likley to be more interesting to get along to (cant defend this strongly, just an incling).

Would like to try to avoid significant feature development to address this, so keen to look at other ways.

Current 'workarounds':

1. Add the repeating event somewhere that alread generates a .ics file and import that file.
The tool mostly supports importing of a .ics file, which obviously could be packed with repeating events, is this helpful?

2. Use the duplicate/clone button. Make it better even?
There's a duplicate/clone on any event. I've been using that to duplicate a monthly gathering forward each month.
What do you think is needed?

Happy to keep discussing.

**Comment by evolve2k** [comment]
I've thought a bit more on recurring events. 
1. It should be easy to add them, there probably should be a date limit, to address the risk of someone adding "Oh this is every week" and then down the track the event is cancelled and its still showing everyone and sort of polluting the feed with its domiance due to its regularity. Gut feel probably can list a years worth or maybe more active would be max of 6 months worth. Upside is getting regular meetings into the calendar. Downside is when a group that meets say month, enevitably changes one of the event dates for some exceptional reason, but because our tool is so seldom used our dates are listed incorrectly. This issue would tend towards people being able to list the next little while but not so far into the future as that they forget about our site and forget to keep it up to date.

2. Just visually the more often an event occurs the more visual domiance it has in the feed. But the paradox is that this event at the same time is less special and important. For example group A is having a special once off guest speaker and there's a listing for that, group B have their regular say fortnightly meeting. Group B's meeting that is always on shouldnt dominant Group A's special event.
I think this may be able to be resolved in that when an event is added as a repeated event, that we can reduce it's visual domance (say smaller font size). Another option is we could have a show/hide for regular events.

**Comment by evolve2k** [comment]
3. In addition to say the show/hide regular/repeating events. Another option is that one of the repeated events a month breaks through the hide into being a showcase. This would give groups some exposure of their regular events without making them mundane. My instinct is to 'reveal/showcase' say the last repeated event in any one month. Meaning regardless of if an event is repeated monthly, fortnightly, weekly or even more often than that, that for all these listing, only one event will show a month in the highlighted main page. The exception might be that we can also show them on the 'today' and 'tomorrow' sections, as its quite nice to know what's on tonight/tomorrow to attend at a whim, and this section for newer ecosystems can be empty more often than not (in my experience).

**Reply by boundingstar** [reply]
I like the idea of being able to show/hide repeating events, although another option i had in mind was to show only the next occurrence of an event so that people always know whats coming up but it doenst crowd up the future calendar, and list the future occurrences/frequency inside the event page. 

For stopping events with incorrect dates, perhaps when creating an event series, the creator also sets reminder time eg. 1 month, where each occurrence will be emailed to them for approval 1 month ahead of when the occurrence is, so that they can check the date, add any other occurrence specific details, and then click create as if it were the regular event creation screen. This would be annoying to people with weekly events though so maybe there can be some way to batch approve a number of upcoming occurrences

**Reply by evolve2k** [reply]
Yeah that's probably a nice approach, just showing the next event probably makes sense, and yes some way when you're in the event to see that there are more upcoming.

**Comment by evolve2k** [comment]
From a UI persepective, I could see the user adding a primary event sort of like normal, and then being able to have a way to say its a repeating event or that it's a series or whatever.

Im thinking the initial assumption could be that the user can add additional events just based on setting the repeating dates, and then this creates a bunch of 'linked' events that can be seen in a listing of like 'other events in this series', which once they are created, could be navigated to, and further edited, offering the ability to say tweak the event title or adjust other details that may vary for a particular occurance of the event.

**Reply by boundingstar** [reply]
yeah makes sense to me, but it would also be good to be able to edit the whole series as well when there's changes to that

**Comment by evolve2k** [comment]
A user just upvoted repeating events as a feature request. They mentioned they like this entry format by Duk Op.

<img width="871" alt="SCR-20250509-nshq" src="https://github.com/user-attachments/assets/31dfaad4-f047-4e06-b393-945efee95be3" />

The tool asks for the interval (as shown) as well as the end date.
This is great from a data entry perspective.

I still have concerns around the visual dominance of regularly repeating items. Also around if an event is added as repeating with an optimistic future timeline (say a 5 year end date) and then the real event stops or has to reschedule. Some risk of our calendar data being unreliable. 

A few remedies could be:
- Users can flag or downvote an event/repeating event and effectivley a couple flags and the repeat is cancelled.
- Only the next couple of months are actually listed, or just the next couple of gatherings.
- The user is notified via email, we're about to list your repeating event for the next month (or whatever), click here if this has changed or is incorrect. Practically this would seem intutive maybe a month at a time, eg in the last week of the month "These repeating events are scheduled in for next month. Any changes, please login and edit. We like to ensure what is listed is actually occuring so we double check repeating events"

**Reply by filipesmedeiros** [reply]
I agree that having tons of repeating events polluting the feed is not good. I would show probably the next instance of the event, and have some indicator that it's recurring (in the details you show the pattern of recurrence).

As for the event changing and becoming stale/abandoned, to be honest I think it can be solved ad-hoc by human moderation. The point of building "small-web"/local tools like Koalagator is that these problems can be socially, and not technically, solved. Either way, a warning system for stale events can always be added later, on top of the first version, no?
